uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-k8s/src/main/java/org/uberfire/java/nio/fs/k8s/K8SFileSystemProvider.java

```diff
@@ -0,0 +1,337 @@
+                                                           true), 
+                             KubernetesClient.class);
+    }
+
+    @Override
+    protected Path[] getDirectoryContent(final Path dirIn) {
+        checkNotNull("dir", dirIn);
+        Path dir = toAbsoluteRealPath(dirIn);
+        if (isRoot(dir) &&
+            !executeCloudFunction(client -> getFsObjCM(client, dir), KubernetesClient.class).isPresent()) {
+            initRoot();
+        }
+        ConfigMap dirCM = executeCloudFunction(client -> getFsObjCM(client, dir), KubernetesClient.class)
+                .orElseThrow(() -> new NotDirectoryException(dir.toString()));
+        if (dirCM.getData() == null || dirCM.getData().isEmpty()) {
+            return new Path[0];
+        }
+        
+        String separator = dir.getFileSystem().getSeparator();
+        String dirPathString = getPathByFsObjCM((K8SFileSystem)fileSystem, dirCM).toString();
+        return dirCM.getData()
+                    .keySet()
+                    .stream()
+                    .map(fileName -> GeneralPathImpl.create(dir.getFileSystem(), 
+                                                           (dirPathString.endsWith(separator) ? 
+                                                            dirPathString :
+                                                            dirPathString.concat(separator)).concat(fileName), 
+                                                            false))
+                    .toArray(Path[]::new);
+    }
+    
+    private synchronized void initRoot() {
+        Path root = this.fileSystem.getPath(K8SFileSystem.UNIX_SEPARATOR_STRING);
+        this.createDirectory(root);
+        logger.info("Root directory created.");
+    }
+
+    @Override
+    public void delete(final Path pathIn, final DeleteOption... options) {
+        checkNotNull("path", pathIn);
+        Path path = toAbsoluteRealPath(pathIn);
+        checkFileNotExistThenThrow(path, false);
+        deleteIfExists(path, options);
+    }
+
+    @Override
+    public boolean deleteIfExists(final Path pathIn, final DeleteOption... options) {
+        checkNotNull("path", pathIn);
+        Path path = toAbsoluteRealPath(pathIn);
+        synchronized (this) {
+            try {
+                return executeCloudFunction(client -> deleteAndUpdateParentCM(client, path), 
+                                            KubernetesClient.class).get();
+            } finally {
+                toGeneralPathImpl(path).clearCache();
+            }
+        }
+    }
+    
+    @Override
+    public boolean isHidden(final Path pathIn) {
+        checkNotNull("path", pathIn);
+        checkFileNotExistThenThrow(pathIn, false);
+        return pathIn.getFileName().toString().startsWith(K8SFileSystemConstants.K8S_FS_HIDDEN_FILE_INDICATOR);
+    }
+
+    @Override
+    public void checkAccess(final Path pathIn, AccessMode... modes) {
```

Comment:
could we please use `final` for modes? just for consistency

Reply:
changed as per suggested